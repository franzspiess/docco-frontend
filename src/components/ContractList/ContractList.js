import { Link } from 'react-router-dom';
/** @jsx jsx */
import { css, jsx } from '@emotion/core';
import styled from '@emotion/styled';
import './ContractList.css';
// import {css,jsx} from '@emotion/core';

export const ContractList = props => {
  const { contractList } = props;

  // renders each contract with font awesome symbol;
  const view = contractList.map(contract => {
    return (
      <div key={contract.id} alt={contract.title} className="my-contract">
        <Link
          css={css`
            text-decoration: none;
            color: #2c3e50;
            pointer: cursor;
          `}
          className="specificContract"
          to={`/contract/${contract.id}`}
        >
          <i
            css={css`
              font-size: 10vh;
            `}
            className="far fa-file"
          />
          <ContractTitle>{contract.title}</ContractTitle>
        </Link>
      </div>
    );
  });

  return <ContractListContainer>{view}</ContractListContainer>;
};

const ContractListContainer = styled('div')`
  display: flex;
  justify-content: flex-start;
  flex-wrap: wrap;
  overflow-y: scroll;
`;
const ContractTitle = styled('div')`
  text-align: center;
  padding-top: 5px;
  width: 150px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

export default ContractList;
