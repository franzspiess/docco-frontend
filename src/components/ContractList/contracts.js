const contracts = [
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' },
  { title: 'My Contract' }
];

export default contracts;
