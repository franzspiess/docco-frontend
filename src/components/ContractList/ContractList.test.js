import React from 'react';
import { shallow, mount } from 'enzyme';
import {ContractList} from './ContractList';




describe.only('<ContractList />', () => {
  let wrapper;
  const mock = [{title:'test'},{title:'test'},{title:'test'},{title:'test'}];
  beforeEach(() => { wrapper = shallow(<ContractList contractList={mock}/>) });
  it('should render ContractList', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('matches the snapshot', () => {
    const tree = mount(<ContractList contractList={mock} />);
    expect(tree).toMatchSnapshot();
    tree.unmount();
  });
  it('should render every contract', () => {
    expect(wrapper.find('.my-contract')).toHaveLength(mock.length);
  });
  it('should render the title', () => {
    expect(wrapper.find('span').at(0).text()).toEqual(mock[0].title);
  });
});