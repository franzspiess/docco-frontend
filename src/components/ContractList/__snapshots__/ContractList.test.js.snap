// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContractList /> matches the snapshot 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <ContractList
    contractList={
      Array [
        Object {
          "title": "test",
        },
        Object {
          "title": "test",
        },
        Object {
          "title": "test",
        },
        Object {
          "title": "test",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "contractList": Array [
        Object {
          "title": "test",
        },
        Object {
          "title": "test",
        },
        Object {
          "title": "test",
        },
        Object {
          "title": "test",
        },
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="contract-list"
      >
        <div
          class="my-contract"
        >
          <i
            class="fas fa-file-contract"
          />
          <span
            class="contract-title"
          >
            test
          </span>
        </div>
        <div
          class="my-contract"
        >
          <i
            class="fas fa-file-contract"
          />
          <span
            class="contract-title"
          >
            test
          </span>
        </div>
        <div
          class="my-contract"
        >
          <i
            class="fas fa-file-contract"
          />
          <span
            class="contract-title"
          >
            test
          </span>
        </div>
        <div
          class="my-contract"
        >
          <i
            class="fas fa-file-contract"
          />
          <span
            class="contract-title"
          >
            test
          </span>
        </div>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="my-contract"
          >
            <i
              className="fas fa-file-contract"
            />
            <span
              className="contract-title"
            >
              test
            </span>
          </div>,
          <div
            className="my-contract"
          >
            <i
              className="fas fa-file-contract"
            />
            <span
              className="contract-title"
            >
              test
            </span>
          </div>,
          <div
            className="my-contract"
          >
            <i
              className="fas fa-file-contract"
            />
            <span
              className="contract-title"
            >
              test
            </span>
          </div>,
          <div
            className="my-contract"
          >
            <i
              className="fas fa-file-contract"
            />
            <span
              className="contract-title"
            >
              test
            </span>
          </div>,
        ],
        "className": "contract-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="my-contract"
          >
            <i
              class="fas fa-file-contract"
            />
            <span
              class="contract-title"
            >
              test
            </span>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-file-contract"
              />,
              <span
                className="contract-title"
              >
                test
              </span>,
            ],
            "className": "my-contract",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <i
                class="fas fa-file-contract"
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-file-contract",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "i",
            },
            Object {
              "instance": <span
                class="contract-title"
              >
                test
              </span>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test",
                "className": "contract-title",
              },
              "ref": null,
              "rendered": Array [
                "test",
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": <div
            class="my-contract"
          >
            <i
              class="fas fa-file-contract"
            />
            <span
              class="contract-title"
            >
              test
            </span>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-file-contract"
              />,
              <span
                className="contract-title"
              >
                test
              </span>,
            ],
            "className": "my-contract",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <i
                class="fas fa-file-contract"
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-file-contract",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "i",
            },
            Object {
              "instance": <span
                class="contract-title"
              >
                test
              </span>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test",
                "className": "contract-title",
              },
              "ref": null,
              "rendered": Array [
                "test",
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": <div
            class="my-contract"
          >
            <i
              class="fas fa-file-contract"
            />
            <span
              class="contract-title"
            >
              test
            </span>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-file-contract"
              />,
              <span
                className="contract-title"
              >
                test
              </span>,
            ],
            "className": "my-contract",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <i
                class="fas fa-file-contract"
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-file-contract",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "i",
            },
            Object {
              "instance": <span
                class="contract-title"
              >
                test
              </span>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test",
                "className": "contract-title",
              },
              "ref": null,
              "rendered": Array [
                "test",
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": <div
            class="my-contract"
          >
            <i
              class="fas fa-file-contract"
            />
            <span
              class="contract-title"
            >
              test
            </span>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-file-contract"
              />,
              <span
                className="contract-title"
              >
                test
              </span>,
            ],
            "className": "my-contract",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <i
                class="fas fa-file-contract"
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-file-contract",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "i",
            },
            Object {
              "instance": <span
                class="contract-title"
              >
                test
              </span>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test",
                "className": "contract-title",
              },
              "ref": null,
              "rendered": Array [
                "test",
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "contractList": Array [
          Object {
            "title": "test",
          },
          Object {
            "title": "test",
          },
          Object {
            "title": "test",
          },
          Object {
            "title": "test",
          },
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="contract-list"
        >
          <div
            class="my-contract"
          >
            <i
              class="fas fa-file-contract"
            />
            <span
              class="contract-title"
            >
              test
            </span>
          </div>
          <div
            class="my-contract"
          >
            <i
              class="fas fa-file-contract"
            />
            <span
              class="contract-title"
            >
              test
            </span>
          </div>
          <div
            class="my-contract"
          >
            <i
              class="fas fa-file-contract"
            />
            <span
              class="contract-title"
            >
              test
            </span>
          </div>
          <div
            class="my-contract"
          >
            <i
              class="fas fa-file-contract"
            />
            <span
              class="contract-title"
            >
              test
            </span>
          </div>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="my-contract"
            >
              <i
                className="fas fa-file-contract"
              />
              <span
                className="contract-title"
              >
                test
              </span>
            </div>,
            <div
              className="my-contract"
            >
              <i
                className="fas fa-file-contract"
              />
              <span
                className="contract-title"
              >
                test
              </span>
            </div>,
            <div
              className="my-contract"
            >
              <i
                className="fas fa-file-contract"
              />
              <span
                className="contract-title"
              >
                test
              </span>
            </div>,
            <div
              className="my-contract"
            >
              <i
                className="fas fa-file-contract"
              />
              <span
                className="contract-title"
              >
                test
              </span>
            </div>,
          ],
          "className": "contract-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="my-contract"
            >
              <i
                class="fas fa-file-contract"
              />
              <span
                class="contract-title"
              >
                test
              </span>
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-file-contract"
                />,
                <span
                  className="contract-title"
                >
                  test
                </span>,
              ],
              "className": "my-contract",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <i
                  class="fas fa-file-contract"
                />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-file-contract",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "i",
              },
              Object {
                "instance": <span
                  class="contract-title"
                >
                  test
                </span>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test",
                  "className": "contract-title",
                },
                "ref": null,
                "rendered": Array [
                  "test",
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": <div
              class="my-contract"
            >
              <i
                class="fas fa-file-contract"
              />
              <span
                class="contract-title"
              >
                test
              </span>
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-file-contract"
                />,
                <span
                  className="contract-title"
                >
                  test
                </span>,
              ],
              "className": "my-contract",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <i
                  class="fas fa-file-contract"
                />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-file-contract",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "i",
              },
              Object {
                "instance": <span
                  class="contract-title"
                >
                  test
                </span>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test",
                  "className": "contract-title",
                },
                "ref": null,
                "rendered": Array [
                  "test",
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": <div
              class="my-contract"
            >
              <i
                class="fas fa-file-contract"
              />
              <span
                class="contract-title"
              >
                test
              </span>
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-file-contract"
                />,
                <span
                  className="contract-title"
                >
                  test
                </span>,
              ],
              "className": "my-contract",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <i
                  class="fas fa-file-contract"
                />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-file-contract",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "i",
              },
              Object {
                "instance": <span
                  class="contract-title"
                >
                  test
                </span>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test",
                  "className": "contract-title",
                },
                "ref": null,
                "rendered": Array [
                  "test",
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": <div
              class="my-contract"
            >
              <i
                class="fas fa-file-contract"
              />
              <span
                class="contract-title"
              >
                test
              </span>
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-file-contract"
                />,
                <span
                  className="contract-title"
                >
                  test
                </span>,
              ],
              "className": "my-contract",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <i
                  class="fas fa-file-contract"
                />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-file-contract",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "i",
              },
              Object {
                "instance": <span
                  class="contract-title"
                >
                  test
                </span>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test",
                  "className": "contract-title",
                },
                "ref": null,
                "rendered": Array [
                  "test",
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
